tests/nested/manual/component-recovery-system/task.yaml

```diff
@@ -0,0 +1,158 @@
+
+    gendeveloper1 sign-model < "${MODEL_JSON}" > model.assert
+
+    cp "${TESTSLIB}/assertions/testrootorg-store.account-key" "${NESTED_FAKESTORE_BLOB_DIR}/asserts"
+    cp "${TESTSLIB}/assertions/developer1.account" "${NESTED_FAKESTORE_BLOB_DIR}/asserts"
+    cp "${TESTSLIB}/assertions/developer1.account-key" "${NESTED_FAKESTORE_BLOB_DIR}/asserts"
+    cp model.assert "${NESTED_FAKESTORE_BLOB_DIR}/asserts"
+
+    tests.nested prepare-essential-snaps
+
+    export SNAPPY_FORCE_API_URL="${NESTED_UBUNTU_IMAGE_SNAPPY_FORCE_SAS_URL}"
+    ubuntu-image snap --channel edge --image-size 10G ./model.assert
+
+    image_dir=$(tests.nested get images-path)
+    image_name=$(tests.nested get image-name core)
+    cp ./pc.img "${image_dir}/${image_name}"
+    tests.nested configure-default-user
+
+    # run the fake device service too, so that the device can be initialised
+    systemd-run --collect --unit fakedevicesvc fakedevicesvc localhost:11029
+
+    tests.nested build-image core
+    tests.nested create-vm core
+
+    #shellcheck source=tests/lib/core-config.sh
+    . "$TESTSLIB"/core-config.sh
+    wait_for_first_boot_change
+
+    remote.exec 'sudo systemctl stop snapd snapd.socket'
+
+    remote.exec 'sudo cat /var/lib/snapd/state.json' | gojq '.data.auth.device."session-macaroon"="fake-session"' > state.json
+    remote.push state.json
+    remote.exec 'sudo mv state.json /var/lib/snapd/state.json'
+    remote.exec 'sudo systemctl start snapd snapd.socket'
+
+restore: |
+    systemctl stop fakedevicesvc
+    "${TESTSTOOLS}/store-state" teardown-fake-store "${NESTED_FAKESTORE_BLOB_DIR}"
+
+execute: |
+  function post_json_data() {
+    route=$1
+    template=$2
+    shift 2
+
+    # shellcheck disable=SC2059
+    response=$(printf "${template}" "$@" | remote.exec "sudo snap debug api -X POST -H 'Content-Type: application/json' ${route}")
+    if ! gojq -e .change <<< "${response}"; then
+      echo "could not get change id from response: ${response}"
+      false
+    fi
+  }
+
+  unsquashfs "${NESTED_FAKESTORE_BLOB_DIR}/pc-kernel.snap"
+  sed -i -e '/^version/ s/$/-with-comps/' squashfs-root/meta/snap.yaml
+  snap pack --filename=pc-kernel-with-comps.snap ./squashfs-root
+  "${TESTSTOOLS}"/build_kernel_with_comps.sh mac80211_hwsim wifi-comp pc-kernel-with-comps.snap
+
+  kernel_id="pYVQrBcKmBa0mZ4CCN7ExT6jH8rY1hza"
+
+  # bump the available kernel version in the fake store
+  "${TESTSTOOLS}"/store-state make-snap-installable --noack \
+    --revision 2 \
+    "${NESTED_FAKESTORE_BLOB_DIR}" \
+    ./pc-kernel-with-comps.snap \
+    "${kernel_id}"
+
+  "${TESTSTOOLS}"/store-state make-component-installable --noack \
+    --snap-revision 2 \
+    --component-revision 1 \
+    --snap-id "${kernel_id}" \
+    "${NESTED_FAKESTORE_BLOB_DIR}" \
+    ./pc-kernel+wifi-comp.comp
+
+  boot_id="$(tests.nested boot-id)"
+  change_id=$(remote.exec "sudo snap refresh --no-wait pc-kernel+wifi-comp")
+  remote.wait-for reboot "${boot_id}"
```

Comment:
This is refreshing the kernel and installing the component, right? Maybe a comment here so we are reminded of this would be good.